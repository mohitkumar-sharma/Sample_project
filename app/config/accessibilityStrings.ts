export default {
  NORMAL_BUTTON_TEXT: 'normal-button-text',
  SLIDER_BUTTON_TEXT: 'slider-button-text',
  NORMAL_BUTTON: 'normal-button',
  SLIDER_BUTTON_ICON: 'slider-button-icon',
  SLIDER_BUTTON_CONTAINER: 'slider-button-container',
  BANNER_TEXT: 'banner-text',
  HEADER_SIDES_TEXT: 'header-sides-text',
  HEADER_PRESSABLE_ELEMENT: 'header-pressable-element',
  SETTINGS_SWITCH_TEXT: 'settings-switch-text',
  SETTINGS_SWITCH: 'settings-switch',
  INV_TEXT_INPUT: 'inv-text-input',
};
