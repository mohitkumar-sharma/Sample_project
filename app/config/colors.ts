export default {
  black: '#000000',
  gray: '#9B9B9B',
  divider: '#E8E8E8',
  white: '#FFFFFF',
  themeBlue: '#6EB1F7',
  themeLightBlack: '#34424A',
  appTheme: '#1A1924',
  borderLine: '#59606A',
  warning: '#FFCC00',
  red: '#FF0000',
};
