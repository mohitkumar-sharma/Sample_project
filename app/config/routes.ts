export default {
  WELCOME: 'Welcome',
  HOME: 'Home',
  SETTINGS: 'Settings',
};
