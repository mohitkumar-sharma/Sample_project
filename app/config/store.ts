import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { rootReducer } from '../states';

const store = configureStore({
  reducer: rootReducer,
  middleware: getDefaultMiddleware(),
});

export type AppDispatch = typeof store.dispatch;

export { store };
