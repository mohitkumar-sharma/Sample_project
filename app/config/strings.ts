export default {
  GO_TO_HOME: 'Go To Home',
  DIAMOND_ICON: 'diamond',
  SLIDE_ME: 'Slide me to continue',
  APP_RUNNING_WARNING: 'Notice: You are running this app on',
  WELCOME_INPUT_PLACEHOLDER: 'Enter your name here...',
  SIGN_OUT: 'SignOut',
  HOME_BUTTONS_HEADING: '4 variations of a button',
  PRESS_ME: 'Press me',
  TRANSPARENT_BTN_PRESSED: 'Transparent button is pressed',
  LIGHT_BLACK_BTN_PRESSED: 'Light black button is pressed',
  BLUE_BTN_PRESSED: 'Blue button is pressed',
  SWITCH_TEXT: 'Enable Warning Banner',
  ALERT_TITLE_INVALID_USER_NAME: 'Invalid UserName!',
  ALERT_TITLE_BTN_PRESSED: 'Button Pressed!',
  ALERT_ENTER_VALID_USER_NAME: 'Please enter a valid user name.',
};
