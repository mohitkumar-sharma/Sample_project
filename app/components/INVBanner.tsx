import React from 'react';
import { View, StyleSheet, StyleProp, ViewStyle, Text, TextStyle } from 'react-native';
import config from '../config';

export interface INVBannerProps {
  /* as a string for Text element to show message on banner */
  text?: string;
  /* as an object of style props for View element */
  containerStyle?: StyleProp<ViewStyle>;
  /* as an object of style props for Text element */
  textStyle?: StyleProp<TextStyle>;
}

/**
 * This is the INVBanner functional component to show the banner in the app
 * @param props : props are defined in INVBannerProps which can be passed to this component
 */
const INVBanner = (props: INVBannerProps): JSX.Element => {
  const { text, textStyle, containerStyle } = props;
  return (
    <View style={[styles.container, containerStyle]}>
      {text && text !== '' ? (
        <Text style={[styles.text, textStyle]} testID={config.accessibilityStrings.BANNER_TEXT}>
          {text}
        </Text>
      ) : null}
    </View>
  );
};

const styles = StyleSheet.create({
  container: { backgroundColor: config.colors.warning },
  text: {
    margin: 10,
    padding: 3,
    fontSize: 15,
    fontWeight: config.fontWeights.FONT_WEIGHT_REGULAR,
    color: config.colors.white,
  },
});

export default INVBanner;
