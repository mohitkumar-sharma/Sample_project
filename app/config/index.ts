import colors from './colors';
import localConfig from './localConfig';
import strings from './strings';
import routes from './routes';
import fontWeights from './fontWeights';
import accessibilityStrings from './accessibilityStrings';
import { store } from './store';

const config = {
  colors,
  ...localConfig,
  store,
  strings,
  routes,
  fontWeights,
  accessibilityStrings,
};

export default config;
