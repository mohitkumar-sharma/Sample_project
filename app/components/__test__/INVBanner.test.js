import React from 'react';
import { render } from '@testing-library/react-native';
import INVBanner from '../INVBanner';
import config from '../../config';

describe('INVBanner', () => {
  const props = {
    textStyle: { fontSize: 17 },
    containerStyle: { backgroundColor: config.colors.red },
  };

  const text = `${config.strings.APP_RUNNING_WARNING} Emulator`;

  it('matches snapshot', () => {
    expect(render(<INVBanner {...props} />)).toMatchSnapshot();
    expect(render(<INVBanner {...props} text={text} />)).toMatchSnapshot();
  });

  it('renders the text', () => {
    const { queryByTestId } = render(<INVBanner {...props} text={text} />);
    expect(queryByTestId(config.accessibilityStrings.BANNER_TEXT)).toBeTruthy();
  });

  it('does not render the text', () => {
    const { queryByTestId } = render(<INVBanner {...props} />);
    expect(queryByTestId(config.accessibilityStrings.BANNER_TEXT)).toBeFalsy();
  });
});
