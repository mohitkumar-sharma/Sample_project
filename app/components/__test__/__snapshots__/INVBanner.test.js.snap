// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`INVBanner matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFCC00",
      },
      Object {
        "backgroundColor": "#FF0000",
      },
    ]
  }
/>
`;

exports[`INVBanner matches snapshot 2`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFCC00",
      },
      Object {
        "backgroundColor": "#FF0000",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 15,
          "fontWeight": "400",
          "margin": 10,
          "padding": 3,
        },
        Object {
          "fontSize": 17,
        },
      ]
    }
    testID="banner-text"
  >
    Notice: You are running this app on Emulator
  </Text>
</View>
`;
